azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java

```diff
@@ -598,43 +602,39 @@ synchronized private Action convertToAction(Status status) {
     // Host the current processing ramp flows
     // Map.Key = FlowId, Map.Value = Set Of Ramps
     private volatile Map<String, Set<String>> executingFlows = new HashMap<>();
-    private volatile int beginFlowCount; // Incoming Ramp Requests After Latest Database Pull Action;
-    private volatile int endFlowCount; // Incoming Ramp Requests After Latest Database Pull Action;
+    private volatile int[] flowCounts = {0, 0}; // [0] count of started flow [1] count of ended flows
```

Comment:
The earlier approach looks cleaner and more readable. Why change it to array?

Reply:
Want to eliminate the potential synchronization problem. I believe volatile int[] way will lock both beginFlowCount and enFlowCount on updating.